rest/aggs/models.go

```diff
@@ -0,0 +1,216 @@
+package aggs
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/polygon-io/client-golang/rest/client"
+)
+
+type Aggregate struct {
+	Ticker            string  `json:"T,omitempty"`
+	Volume            float64 `json:"v"`
+	VWAP              float64 `json:"vw,omitempty"`
+	AggregateVWAP     float64 `json:"a,omitempty"`
+	Open              float64 `json:"o"`
+	Close             float64 `json:"c"`
+	High              float64 `json:"h"`
+	Low               float64 `json:"l"`
+	Timestamp         int64   `json:"t"`
+	Transactions      int64   `json:"n,omitempty"`
+	Market            string  `json:"m,omitempty"`
+	Exchange          int32   `json:"x,omitempty"`
+	Locale            string  `json:"g,omitempty"`
+	OfficialOpenPrice float64 `json:"op,omitempty"`
+	AverageSize       float64 `json:"z,omitempty"`
+	AccumulatedVolume float64 `json:"av,omitempty"`
+	StartTimestamp    int64   `json:"s,omitempty"`
+	EndTimestamp      int64   `json:"e,omitempty"`
+}
+
+type AggsResponse struct {
+	client.BaseResponse
+	Ticker       string      `json:"ticker,omitempty"`
+	QueryCount   int         `json:"queryCount"`
+	ResultsCount int         `json:"resultsCount"`
+	Adjusted     bool        `json:"adjusted"`
+	Aggs         []Aggregate `json:"results,omitempty"`
+}
+
+func (r *AggsResponse) UnmarshalJSON(data []byte) error {
+	type aggsResponse AggsResponse
+	var v aggsResponse
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	*r = AggsResponse(v)
+	return nil
+}
+
+type GetParams struct {
+	Ticker      string
+	Multiplier  int
+	Resolution  string
+	From        time.Time
+	To          time.Time
+	QueryParams *GetQueryParams
```

Comment:
Not every endpoint has path params, but the ones that do have them require them.

Constructing the params as-is looks like this:
```
	res, err := c.Aggs.GetGroupedDaily(context.Background(), aggs.GetGroupedDailyParams{
		Locale:     "global",
		MarketType: aggs.Crypto,
		Date:       time.Date(2021, 7, 21, 0, 0, 0, 0, time.Local),
		QueryParams: &aggs.GetGroupedDailyQueryParams{
			Adjusted: true,
		},
	})
```

If `Get{API}Params` had a struct for the path params, constructing that same request would look like this:

```
	res, err := c.Aggs.GetGroupedDaily(context.Background(), aggs.GetGroupedDailyParams{
		PathParams: {
                 	Locale:     "global",
			MarketType: aggs.Crypto,
			Date:       time.Date(2021, 7, 21, 0, 0, 0, 0, time.Local),
		},
		QueryParams: &aggs.GetGroupedDailyQueryParams{
			Adjusted: true,
		},
	})
```

It's pretty redundant to put the path params in the PathParams struct since they are already required when creating any request to this endpoint. None of the fields inside of PathParams can be left out.

For these reasons we decided to simplify constructing each query without `PathParams`.

This pattern works for each endpoint, so our docs can state 
```
All path params are required and are listed as `Get{API}Params` fields except for the optional `QueryParams` 
```